src/browser/iframe.js

```diff
@@ -92,7 +137,22 @@
       });
     })
   });
-
-  // Exports:
-
   exports.view = view;
+
+  // Gecko API for handling `window.open` has very unfortunate design
+  // (see #566, #565, #564). To workaround it and present more desirable
+  // API that also works with react we employ a hack to generate `opener`
+  // `String` instance (not a primitive value) that holds a reference to
+  // an iframe. It is used to box / unbox actual iframe that is then used
+  // by an IFrame to replace nodes.
+  const Opener = iframe => {
+    // See: https://github.com/mozilla/browser.html/issues/568
+    iframe.isSetVisibleBroken = true;
```

Comment:
You can also wrap setVisible in a try/catch statement as well - whichever you feel better about.